backend/src/main/java/com/example/backend/service/AuthenticationService.java

```diff
@@ -0,0 +1,208 @@
+                                + "Thank you,<br>"
+                                + "TSA.";
+
+                MimeMessage message = mailSender.createMimeMessage();
+                MimeMessageHelper helper = new MimeMessageHelper(message, true, "UTF-8");
+
+                helper.setFrom(fromAddress, senderName);
+                helper.setTo(toAddress);
+                helper.setSubject(subject);
+
+                content = content.replace("[[name]]", user.getUsername());
+                String verifyURL = siteURL + "/api/v1/auth/verify?code=" + user.getVerificationToken();
+
+                content = content.replace("[[URL]]", verifyURL);
+
+                helper.setText(content, true);
+
+                mailSender.send(message);
+        }
+
+        public boolean verifyUser(String verificationToken) {
+                UserModel user = userRepository.findByVerificationToken(verificationToken).orElse(null);
+                if (user == null || user.isEnabled()) {
+                        return false;
+                }
+
+                // enable user
+                user.setEnabled(true);
+                userRepository.save(user);
+
+                return true;
+
+        }
+
+        public AuthenticationResponse register(RegisterRequest request) {
+                // verify that the username and email are unique
+                UserModel userByUsername = userRepository.findByUsername(request.getUsername()).orElse(null);
+                UserModel userByEmail = userRepository.findByEmail(request.getEmail()).orElse(null);
+
+                // user already exists => conflict
+                if (userByEmail != null || userByUsername != null) {
+                        return AuthenticationResponse.builder()
+                                        .status(HttpStatus.CONFLICT)
+                                        .build();
+                }
+
+                // generate random string for the verification token
+                String randomToken = RandomString.make(64);
+
+                // build user model
+                UserModel user = UserModel.builder()
+                                .username(request.getUsername())
+                                .email(request.getEmail())
+                                .password(passwordEncoder.encode(request.getPassword()))
+                                .role("USER")
+                                .enabled(false)
+                                .verificationToken(randomToken)
+                                .build();
+
+                // save in db
+                userRepository.save(user);
+
+                // send verification email and handle exceptions
+                try {
+                        sendVerificationEmail(user);
+                } catch (UnsupportedEncodingException | MessagingException e) {
+                        System.out.println("Error while sending verification email.");
```

Comment:
Should we send a message to frontend to tell that the sending failed? (not sure)